mlx/backend/common/default_primitives.cpp

```diff
@@ -190,4 +224,140 @@ void AddMM::eval_cpu(const std::vector<array>& inputs, array& out) {
   return matmul_common_general(inputs[0], inputs[1], out, alpha_, beta_);
 }
 
+void BlockMaskedMM::eval_cpu(const std::vector<array>& inputs, array& out) {
```

Comment:
Maybe I'm missing something, but is this identical to the `eval_cpu` for the accelerate backend? If so I would refactor to just use this one. You can delete that on and move the implementation to a C++ file which is not conditionally included.

Reply:
Oh yes, I can do that for now, but later so that we can do `float16` and `bfloat16`, the same logic would be used for for the BNNS calls - so I ended up making copies

Reply:
Fair enough. Maybe we should have this be in a `BlockMaskedMM::eval` function to avoid duplicating all that code and dispatch to it for fp32?